File: healthcare_management/src/main/java/org/example/healthcare_management/entities/Clinic.java
package org.example.healthcare_management.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "clinics") // bảng phòng khám
public class Clinic {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "address")
    private String address;

    @Column(name = "phone")
    private String phone;

    // mappedBy trỏ tới tên biến clinic trong entity Doctor
    @OneToMany(
            mappedBy = "clinic",
            cascade = {
                    CascadeType.PERSIST,
                    CascadeType.MERGE,
                    CascadeType.REFRESH,
                    CascadeType.DETACH
            }, fetch = FetchType.LAZY)
    private Set<Doctor> doctors = new HashSet<>();

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "image")
    private String image;

    @Column(name = "createdAt")
    private LocalDateTime createdAt;

    @Column(name = "updatedAt")
    private LocalDateTime updatedAt;

    @Column(name = "deletedAt")
    private LocalDateTime deletedAt;

}

File: healthcare_management_app/lib/models/Clinic.dart
class Clinic {
  Clinic({
    required this.id,
    required this.name,
    required this.address,
    required this.phone,
    required this.description,
    required this.image,
    required this.createdAt,
     this.updatedAt,
    this.deletedAt,
  });

  final int? id;
  final String name;
  final String? address;
  final String? phone;
  final dynamic description;
  final dynamic image;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final DateTime? deletedAt;

  factory Clinic.fromJson(Map<String, dynamic> json){
    return Clinic(
      id: json["id"],
      name: json["name"],
      address: json["address"],
      phone: json["phone"],
      description: json["description"],
      image: json["image"],
      createdAt: DateTime.tryParse(json["createdAt"] ?? ""),
      updatedAt: DateTime.tryParse(json["updatedAt"] ?? ""),
      deletedAt: DateTime.tryParse(json["deletedAt"] ?? ""),
    );
  }

  Map<String, dynamic> toJson() => {
    "id": id,
    "name": name,
    "address": address,
    "phone": phone,
    "description": description,
    "image": image,
    "createdAt": createdAt?.toIso8601String(),
    "updatedAt": updatedAt?.toIso8601String(),
    "deleteAt": deletedAt?.toIso8601String(),
  };

  @override
  String toString(){
    return "$id, $name, $address, $phone, $description, $image, $createdAt,$updatedAt,$deletedAt";
  }
}
